.github/workflows/continous-integration-docker.yml

```
@@ -255,10 +318,18 @@ jobs:
       zebra_state_dir: 'zebrad-cache'
 
   # Test that Zebra can handle a lightwalletd send transaction RPC call, using a cached Zebra tip state
+  #
+  # Runs:
+  # - after every PR is merged to `main`
+  # - on every PR update
+  #
+  # If the state version has changed, waits for the new cached states to be created.
+  # Otherwise, if the state rebuild was skipped, runs immediately after the build job.
+  # TODO: move this job under lightwalletd-full-sync to have a sequential logic
   lightwalletd-transactions-test:
-    needs: build
+    needs: [ test-full-sync, lightwalletd-full-sync ]
```

nitpick: this dependency is redundant, because `lightwalletd-full-sync` depends on `test-full-sync`

```suggestion
    needs: lightwalletd-full-sync
```